test cases/format/1 default/gh13242.meson

```diff
@@ -12,5 +12,7 @@ test(
 test(
     should_fail: (settings.get('x', false) and not settings['y'] and dep.version().version_compare(
         '>=1.2.3',
-    )),
+    )
```

Comment:
This change does modify the behavior of this test case, although it's not clear to me if this test case really wants the formatting it has or if it is there just to detect a bug.

A similar tool like black would end up formatting this block like:

```python
test(
    should_fail=(
        settings.get("x", false)
        and not settings["y"]
        and dep.version().version_compare(
            ">=1.2.3",
        )
    ),
)
```

Reply:
I think that the right parenthesis should not be on a separate line if the left parenthesis is not a a separate line.

If I put `if node.lpar.whitespaces and '\n' in node.lpar.whitespaces.value:` just before `self.add_nl_after(node.inner)`, it seems to fix it.